### PR TITLE
Implemented Signup, Login, and OTP Pages 

### PR DESCRIPTION
**Pull request by NidhiJain08**
Created the Signup, Login, and OTP pages as per the specifications mentioned in the Figma design.
The OTP page can be accessed at the route: `/otppage`

### CODE REVIEW

File: src/pages/Login.jsx [hunk 1]
```diff
@@ -1,75 +1,205 @@
...
-import google from "../assets/svg/google.svg"
+import { useState,useRef,useEffect } from 'react';
+import "../assets/styles/Login.css";
 import Navbar from "../components/Navbar.jsx";
-import Footer from "../components/Footer.jsx";
+import { Link } from 'react-router-dom';
+import SignUp from './SignUp.jsx';
+
+const USER_REGEX = /^[A-Za-z][A-Za-z0-9_]{3,29}$/; // 4-30 characters
+const PWD_REGEX = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%]).{8,24}$/;
```
Comment (ajaynegi45) [inline src/pages/Login.jsx:9]: Improve this regex for password

File: src/pages/Login.jsx [hunk 2]
```diff
@@ -1,75 +1,205 @@
...
+                        <button>Github</button>
+                        <button>LinkedIn</button>
+                    </div>
+                </div>
+            </form>
+
+            <p>
+                Don't have an account?             
+                   <Link to="/signup" > Sign Up </Link>
+            </p>
```
Comment (ajaynegi45) [inline src/pages/Login.jsx:109]: Make the whole sentence link

File: src/pages/OtpPage.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import "../assets/styles/OtpPage.css";
+
+const OtpPage = () => {
+  return (
+    <div className="otp-container">
+      <h1>One-Time Password</h1>
+      <div className="otp-inputs">
+        {[...Array(6)].map((_, index) => (
+          <input
```
Comment (ajaynegi45) [inline src/pages/OtpPage.jsx:10]: After entering the digit the focus or active input shift to next.

**Review by ajaynegi45 — CHANGES_REQUESTED** [review]
Hi @NidhiJain08,

Thank you for your prompt work on this issue. I appreciate your dedication to the project.

I have reviewed your work, and I need some changes which are mentioned in code comment.

Once again, thank you for your contribution! ❤️

File: src/pages/Login.jsx (hunk 1 above)
Comment (NidhiJain08) [reply src/pages/Login.jsx:9]: Can you explain what changes can be done to improve it?

**Review by Devamani11D — CHANGES_REQUESTED** [review]
You can add following checks for password:
1.Check if there are any common patterns in passwords such as 123456,password as passwords.
2. Check if any character is repeated more than twice at a time.

**Comment by NidhiJain08** [comment]
Hi @Devamani11D ,
I have now done the changes as per requirement.
Thank You!

**Review by ajaynegi45 — APPROVED** [review]
Hi @NidhiJain08,

Thank you for your prompt work on this issue. I appreciate your dedication to the project.

I have reviewed your work, and I'm pleased to inform you that I have merged your pull request. Your changes looks fantastic, and it aligns perfectly with the project's design and usability goals. Great job!

If you have any more contributions or if there's anything else you'd like to work on in the future, please don't hesitate to let us know. Your efforts are valuable to the project's success.

Once again, thank you for your contribution! ❤️

**Comment by NidhiJain08** [comment]
> Hi @NidhiJain08,
> 
> Thank you for your prompt work on this issue. I appreciate your dedication to the project.
> 
> I have reviewed your work, and I'm pleased to inform you that I have merged your pull request. Your changes looks fantastic, and it aligns perfectly with the project's design and usability goals. Great job!
> 
> If you have any more contributions or if there's anything else you'd like to work on in the future, please don't hesitate to let us know. Your efforts are valuable to the project's success.
> 
> Once again, thank you for your contribution! ❤️

Hi @ajaynegi45 ,
Thank you for merging the request, I am glad that you liked the changes.
Thanks again for giving me the opportunity.

**Pull request merged**